LLMonFHIR/FHIR Display/OpenAIChatView.swift

```diff
@@ -87,4 +77,109 @@ struct OpenAIChatView: View {
             }
         }
     }
+    
+    private func processMultipleResourceChat() async throws {
+        let resourcesArray = await fhirStandard.resources
+        var stringResourcesArray = resourcesArray.map { "\($0.displayName) in \($0.resourceType)" }
+        stringResourcesArray.append("N/A")
+        let functionCallOutputArray = try await getFunctionCallOutputArray(stringResourcesArray)
+        
+        if !functionCallOutputArray.contains(where: { $0 == "N/A" }) && !functionCallOutputArray.contains(where: { $0 == " N/A" }) {
+            processFunctionCallOutputArray(functionCallOutputArray: functionCallOutputArray, resourcesArray: resourcesArray)
+        }
+    }
+
+    private func getFunctionCallOutputArray(_ stringResourcesArray: [String]) async throws -> [String] {
+        let openAI = OpenAI(apiToken: openAPIComponent.apiToken ?? "")
+        
+        let functions = [
+            ChatFunctionDeclaration(
+                name: "get_resource_titles",
+                description: String(localized: "MULTIPLE_RESOURCE_FUNCTION_DESCRIPTION"),
+                parameters: JSONSchema(
+                    type: .object,
+                    properties: [
+                        "resources": .init(
+                            type: .string,
+                            description: String(localized: "MULTIPLE_RESOURCE_PARAMETER_DESCRIPTION"),
+                            enumValues: stringResourcesArray
+                        )
+                    ],
+                    required: ["resources"]
+                )
+            )
+        ]
+        
+        let query = ChatQuery(
+            model: "gpt-3.5-turbo-0613",
+            messages: [
+                Chat(role: .user, content: chat.last?.content ?? ""),
+                Chat(role: .system, content: String(localized: "MULTIPLE_RESOURCE_FUNCTION_CONTEXT") + stringResourcesArray.rawValue)
+            ],
```

Comment:
I foresee one problems with the current approach: We loose all the context every time we make a function call as this only contains the last message from the chat, it would be wise if we always send the complete chat when making any API request to the model. This allows it to keep the complete conversation history in mind when answering questions and e.g. only request new resources it needs access to.

Therefore I would suggest the following procedure:
1. Let's append the function call to the chat when the chat is first passed to the view (e.g. in the initializer).
2. Always use the current chat instance to make the request, this always contains the information that the function call is a possible option for the model and it can choose to invoke it at any time.

This would also dramatically reduce the different between the multiple resources chat and the current setup: The only different would be if we append the functions call message in the initializer. If we do that, then the model can use that and might at some point respond with a function call.
We can then check this in both use cases (it will only happen in the multipel resources chat one) and therefore reuse a lot of the code across the two use cases and don't need to make this a big difference path.

What do you think about that idea?